.github/workflows/ci.yml

```
@@ -0,0 +1,100 @@
+
+    - name: Run type check
+      run: npx tsc --noEmit
+
+    - name: Run tests
+      run: npm run test:ci
+
+    - name: Build project
+      run: npm run build
+
+    - name: Upload coverage reports
+      uses: codecov/codecov-action@v3
```

actionlint (1.7.7)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)